src/main/kotlin/io/github/gradlenexus/publishplugin/FindStagingRepository.kt

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin
+
+import io.github.gradlenexus.publishplugin.internal.NexusClient
+import io.github.gradlenexus.publishplugin.internal.StagingRepositoryDescriptorRegistry
+import org.gradle.api.GradleException
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.provider.Provider
+import org.gradle.api.tasks.Input
+import org.gradle.api.tasks.Internal
+import org.gradle.api.tasks.Optional
+import org.gradle.api.tasks.TaskAction
+import org.gradle.kotlin.dsl.property
+import javax.inject.Inject
+
+open class FindStagingRepository @Inject constructor(
+    objects: ObjectFactory,
+    extension: NexusPublishExtension,
+    repository: NexusRepository,
+    private val registry: Provider<StagingRepositoryDescriptorRegistry>
+) : AbstractNexusStagingRepositoryTask(objects, extension, repository) {
+
+    @Optional
+    @Input
+    val packageGroup = objects.property<String>().apply {
+        set(extension.packageGroup)
+    }
+
+    @Input
+    val descriptionRegex = objects.property<String>().apply {
+        set(extension.repositoryDescription.map { "\b" + Regex.escape(it) + "(\\s|$)" })
+    }
+
+    @Internal
+    val stagingRepositoryId = objects.property<String>()
+
+    init {
+        outputs.cacheIf("the task requests data from the external repository, so we don't want to cache it") {
+            false
+        }
+    }
+
+    @TaskAction
+    fun findStagingRepository() {
+        val repository = repository.get()
+        val serverUrl = repository.nexusUrl.get()
+        val client = NexusClient(serverUrl, repository.username.orNull, repository.password.orNull, clientTimeout.orNull, connectTimeout.orNull)
+        val stagingProfileId = determineStagingProfileId(repository, client)
+        logger.info("Fetching staging repositories for {} at {}, stagingProfileId '{}'", repository.name, serverUrl, stagingProfileId)
+        val descriptionRegex = descriptionRegex.get()
+        val descriptor = client.findStagingRepository(stagingProfileId, Regex(descriptionRegex))
+        logger.lifecycle("Staging repository for {} at {}, stagingProfileId '{}', descriptionRegex '{}' is '{}'", repository.name, serverUrl, stagingProfileId, descriptionRegex, stagingRepositoryId)
+        this.stagingRepositoryId.set(stagingRepositoryId)
```

Comment:
It seems, there is something wrong here. The output is:
```
> Task :findSonatypeStagingRepository
Staging repository for sonatype at http://localhost:35705, stagingProfileId 'someProfileId', descriptionRegex '\Qorg.example:sample:0.0.1\E(\s|$)' is task ':findSonatypeStagingRepository' property 'stagingRepositoryId'
```

And in fact the logged uses the input which itself is set in the next line (however, `this.stagingRepositoryId` and `stagingRepositoryId` seems to point to the same object/property).